Add RBAC for tink-controller

abhinavmpandey08 opened the pull request:
## Description

<!--- Please describe what this PR is going to change -->

## Why is this needed

<!--- Link to issue you have raised -->

Fixes: #

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->


## How are existing users impacted? What migration steps/scripts do we need?

<!--- Fixes a bug, unblocks installation, removes a component of the stack etc -->
<!--- Requires a DB migration script, etc. -->


## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

displague committed:
error rather than panic when workflow action index is invalid

Signed-off-by: Marques Johansson <mjohansson@equinix.com>

abhinavmpandey08 committed:
Add RBAC for tink-controller and tink-server

Signed-off-by: Abhinav Pandey <abhinavmpandey08@gmail.com>

micahhausler commented on config/rbac/tink-server/role.yaml:
```diff
@@ -0,0 +1,34 @@
+    verbs:
+      - get
+      - list
+      - watch
```
server will need mutating permissions on workflows

abhinavmpandey08 replied:
added

mmlb committed:
Krm/tink server (#573)

## Description

This PR adds a Kubernetes API backend to Tinkerell as a replacement for Postgres.

This PR builds on #563 and adds new API capabilities

## Why is this needed

## How Has This Been Tested?

## How are existing users impacted? What migration steps/scripts do we need?

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

jacobweinstock committed:
Add repo roles:

This is needed to enable the repo roles defined by
the community here: tinkerbell/org#10

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

mergify[bot] committed:
Add repo roles: (#611)

## Description


This is needed to satisfy the repo roles defined by the community here: https://github.com/tinkerbell/org/issues/10

## Why is this needed



Fixes: #

## How Has This Been Tested?





## How are existing users impacted? What migration steps/scripts do we need?





## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

gauravgahlot committed:
👌 IMPROVE: resolve conflict

Signed-off-by: Gaurav Gahlot <gauravgahlot0107@gmail.com>

gauravgahlot committed:
👌 IMPROVE: check if the total number of actions is more than 1

Signed-off-by: Gaurav Gahlot <gauravgahlot0107@gmail.com>
